src/Storage/AuthorizationCodeStorage.php

```diff
@@ -85,4 +89,29 @@ public function expireAuthorizationCode( $code ) {
 			delete_user_meta( $user_id, self::META_KEY_PREFIX . '_' . $key . '_' . $code );
 		}
 	}
+
+	/**
+	 * This function cleans up auth codes that are sitting in the database because of interrupted/abandoned OAuth flows.
+	 */
+	public function cleanupOldCodes() {
+		global $wpdb;
+
+		// phpcs:ignore WordPress.DB.DirectDatabaseQuery.DirectQuery,WordPress.DB.DirectDatabaseQuery.NoCaching
+		$data = $wpdb->get_results( "SELECT user_id, meta_key, meta_value FROM $wpdb->usermeta WHERE meta_key LIKE 'oidc_expires_%';" );
```

Comment:
I am not a fan of a direct db query but this would give us the chance to query for the expiry value so that we don't need to do the comparison in PHP. And please: no semicolons in queries.

```suggestion
		$data = $wpdb->get_results( $wpdb->prepare( "SELECT user_id, meta_key FROM $wpdb->usermeta WHERE meta_key LIKE 'oidc_expires_%' AND meta_value < %d", time()- 3600 ) );
```

Reply:
I purposely kept the query only having meta_key column in WHERE clause. Would this not impact SQL query & just add another filtration on rows after first WHERE clause?

Also curious, why semicolons in queries should be avoided?

Reply:
> I purposely kept the query only having meta_key column in WHERE clause. Would this not impact SQL query & just add another filtration on rows after first WHERE clause?

It'd would require less data to be transferred to PHP. It is traversing the rows anyway so it can easily filter it out alongside.

Reply:
Gotcha, thanks!